### PR TITLE
feat: impl drop view

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
#3729 
## What's changed and what's your intention?

Supports `DROP VIEW` statement:

```sql
mysql> CREATE VIEW v1 as SELECT * FROM numbers LIMIT 10;
Query OK, 0 rows affected (0.03 sec)

mysql> SELECT * FROM v1;
+--------+
| number |
+--------+
|      0 |
|      1 |
|      2 |
|      3 |
|      4 |
|      5 |
|      6 |
|      7 |
|      8 |
|      9 |
+--------+
10 rows in set (0.02 sec)

mysql> DROP VIEW v1;
Query OK, 0 rows affected (0.01 sec)

mysql> SELECT * FROM v1;
ERROR 1815 (HY000): Failed to plan SQL: Error during planning: Table not found: greptime.public.v1
```

Main changes:

* Adds `DropViewProcedure` to drop view.
* Implements parser for `DROP VIEW [IF EXISTS] view` statement.
* Updated the sqlness tests to use `drop view`.

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Implemented the ability to drop views in the database system.
  - Enhanced error handling with new error variants and updated match patterns.
  - Added new metrics for drop view procedures.

- **Bug Fixes**
  - Corrected SQL scripts and test cases to align with new view-dropping functionality.

- **Refactor**
  - Updated constant values in the `DropFlowProcedure` struct to ensure consistency.

- **Tests**
  - Added comprehensive test cases for dropping views, including scenarios for non-existent views and successful drops.

### CODE REVIEW

File: tests/cases/standalone/common/view/show_create.sql [hunk 1]
```diff
@@ -38,7 +34,8 @@ SHOW CREATE VIEW v1;
 
 SELECT * FROM v1;
 
+DROP VIEW v1;
```
[the remark this answers is not in the document]
Comment (killme2008) [reply tests/cases/standalone/common/view/show_create.sql:37]: The SQL located in `tests/cases` represents the test cases designed to target specific corner cases, so the suggestion doesn't make sense, please remember it.

File: src/common/meta/src/ddl/drop_view.rs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+                task,
+            },
+        }
+    }
+
+    pub fn from_json(json: &str, context: DdlContext) -> ProcedureResult<Self> {
+        let data: DropViewData = serde_json::from_str(json).context(FromJsonSnafu)?;
+
+        Ok(Self { context, data })
+    }
```
[the remark this answers is not in the document]
Comment (killme2008) [reply src/common/meta/src/ddl/drop_view.rs:63]: No, I think the current behavior is more graceful.

File: src/common/meta/src/ddl/drop_view.rs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            error::ViewNotFoundSnafu {
+                view_name: table_ref.to_string(),
+            }
+        );
+
+        self.check_view_metadata().await?;
+        self.data.state = DropViewState::DeleteMetadata;
+
+        Ok(Status::executing(true))
+    }
```
[the remark this answers is not in the document]
Comment (killme2008) [reply src/common/meta/src/ddl/drop_view.rs:102]: The `error` enum has the display function which will provide more context, so please don't give these suggestions anymore.

File: src/common/meta/src/ddl/drop_view.rs [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+                &ctx,
+                &[
+                    CacheIdent::TableId(view_id),
+                    CacheIdent::TableName(self.data.table_ref().into()),
+                ],
+            )
+            .await?;
+
+        Ok(Status::done())
+    }
```
[the remark this answers is not in the document]
Comment (killme2008) [reply src/common/meta/src/ddl/drop_view.rs:170]: Doesn't make sense.

File: src/common/meta/src/ddl/drop_view.rs [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        let view_id = self.data.view_id();
+        let lock_key = vec![
+            CatalogLock::Read(table_ref.catalog).into(),
+            SchemaLock::read(table_ref.catalog, table_ref.schema).into(),
+            TableLock::Write(view_id).into(),
+        ];
+
+        LockKey::new(lock_key)
+    }
+}
```
[the remark this answers is not in the document]
Comment (killme2008) [reply src/common/meta/src/ddl/drop_view.rs:208]: Doesn't make sense.

**Review by WenyXu — APPROVED** [review]
LGTM

File: src/common/meta/src/ddl/drop_view.rs [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+/// The procedure for dropping a view.
+pub struct DropViewProcedure {
+    /// The context of procedure runtime.
+    pub(crate) context: DdlContext,
+    /// The serializable data.
+    pub(crate) data: DropViewData,
+}
+
+impl DropViewProcedure {
+    pub const TYPE_NAME: &'static str = "metasrv-procedure:DropView";
```
Comment (fengjiachun) [inline src/common/meta/src/ddl/drop_view.rs:46]:
```suggestion
    pub const TYPE_NAME: &'static str = "metasrv-procedure::DropView";
```

File: src/common/meta/src/rpc/ddl.rs [hunk 7]
```diff
@@ -80,6 +88,7 @@ impl DdlTask {
         DdlTask::CreateTable(CreateTableTask::new(expr, partitions, table_info))
     }
 
+    /// Creates a [`DdlTask`] to create a logical table.
```
Comment (fengjiachun) [inline src/common/meta/src/rpc/ddl.rs:91]:
```suggestion
    /// Creates a [`DdlTask`] to create several logical tables.
```

File: src/common/meta/src/rpc/ddl.rs [hunk 8]
```diff
@@ -89,6 +98,7 @@ impl DdlTask {
         )
     }
 
+    /// Creates a [`DdlTask`] to alter a logical table.
```
Comment (fengjiachun) [inline src/common/meta/src/rpc/ddl.rs:101]:
```suggestion
    /// Creates a [`DdlTask`] to alter several logical tables.
```

**Review by fengjiachun — APPROVED** [review]
LGTM
